### PR TITLE
Adds power cells and cell charger

### PR DESCRIPTION
**Pull request by LightVelox**
It adds the power cells and the cell charger for future usage with machinery, energy guns, etc...
Also a "Take Store" Interaction for things that only have 1 slot

### CODE REVIEW

File: Assets/Content/Furniture/Machines/Science/CellCharger.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+{
+    private StorageContainer storageContainer;
+    private Container container;
+    private Item powerCell;
+    public MeshRenderer renderer;
+
+    private void Start()
+    {
+        container = GetComponent<Container>();
+        StartCoroutine("StartCharge");
```
Comment (Alainx277) [inline Assets/Content/Furniture/Machines/Science/CellCharger.cs:23]: This should be wrapped in an isServer check.
```suggestion
        if (isServer)
        {
            StartCoroutine(StartCharge());
        }
```

File: Assets/Content/Furniture/Machines/Science/CellCharger.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    private void Update()
+    {
+        powerCell = container.GetItem(0);
```
Comment (Alainx277) [inline Assets/Content/Furniture/Machines/Science/CellCharger.cs:28]: Consider using the onChange event on the container for this.

File: Assets/Content/Items/Functional/Generic/Power Cells/PowerSupply.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            this.charge = charge;
+            this.maxCharge = maxCharge;
+            this.chargeRate = chargeRate;
+            this.selfRecharge = selfRecharge;
+        }
+
+        public PowerSupply WithCharge(int ccharge)
+        {
+            return new PowerSupply(ccharge, maxCharge, chargeRate, selfRecharge);
+        }
```
Comment (Alainx277) [inline Assets/Content/Items/Functional/Generic/Power Cells/PowerSupply.cs:31]: Is there a reason why this creates a new instance?

File: Assets/Engine/Inventory/Extensions/TakeStoreInteraction.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+using SS3D.Engine.Interactions;
+using SS3D.Engine.Interactions.Extensions;
+using SS3D.Engine.Inventory;
+using SS3D.Engine.Inventory.Extensions;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SS3D.Content.Systems.Interactions
+{
+    public class Store : StoreInteraction { };
```
Comment (Alainx277) [inline Assets/Engine/Inventory/Extensions/TakeStoreInteraction.cs:11]: Please use `using Store = SS3D.Content.Systems.Interactions.StoreInteraction`

File: Assets/Engine/Inventory/Extensions/TakeStoreInteraction.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        private void Take(InteractionEvent interactionEvent, InteractionReference arg2)
+        {
+            Hands hands = (Hands)interactionEvent.Source.GetRootSource();
+            hands.GameObject.GetComponent<Inventory>()
+                            .MoveItem(((IGameObjectProvider)interactionEvent.Target).GameObject, 0, hands.ContainerObject, hands.HeldSlot);
+        }
+
+        private bool CanTake(InteractionEvent interactionEvent)
+        {
+            return ((IGameObjectProvider)interactionEvent.Target).GameObject.GetComponent<Container>().GetItem(0) != null;
```
Comment (Alainx277) [inline Assets/Engine/Inventory/Extensions/TakeStoreInteraction.cs:40]: You should check if the source is hands and not holding anything.

File: Assets/Engine/Inventory/Extensions/TakeStoreInteraction.cs (hunk 4 above)
Comment (LightVelox) [reply Assets/Engine/Inventory/Extensions/TakeStoreInteraction.cs:11]: I'll have to change the StoreInteraction so you can only store a Power Cell on it

Comment (Alainx277) [reply Assets/Engine/Inventory/Extensions/TakeStoreInteraction.cs:11]: It's perfectly fine then.

File: Assets/Content/Items/Functional/Generic/Power Cells/PowerSupply.cs (hunk 3 above)
Comment (LightVelox) [reply Assets/Content/Items/Functional/Generic/Power Cells/PowerSupply.cs:31]: It's a struct, you can't call it without the new keyword

Comment (Alainx277) [reply Assets/Content/Items/Functional/Generic/Power Cells/PowerSupply.cs:31]: I don't mean the new keyword, why doesn't it just change the value?

Comment (LightVelox) [reply Assets/Content/Items/Functional/Generic/Power Cells/PowerSupply.cs:31]: cause this is not used only for changing the charge, it's also used to create a power cell with an charge already

**Pull request merged**